src/libstd/keyword_docs.rs

```diff
@@ -1578,12 +1578,255 @@ mod type_keyword {}
 
 #[doc(keyword = "unsafe")]
 //
-/// Code or interfaces whose [memory safety] cannot be verified by the type system.
+/// Code or interfaces whose [memory safety] cannot be verified by the type
+/// system.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `unsafe` keyword has two uses: to declare the existence of contracts the
+/// compiler can't check (`unsafe fn` and `unsafe trait`), and to declare that a
+/// programmer has checked that these contracts have been upheld (`unsafe {}`
+/// and `unsafe impl`, but also `unsafe fn` -- see below). They are not mutually
+/// exclusive, as can be seen in `unsafe fn`.
+///
+/// # Unsafe abilities
+///
+/// Unsafe Rust adds some capabilities to Safe Rust:
+///
+/// - Dereference [raw pointers]
+/// - Implement `unsafe` [`trait`]s
+/// - Call `unsafe` functions (including foreign functions, some compiler
+/// intrinsics and the raw allocator)
+/// - Mutate [`static`]s (including [`extern`]al ones)
+/// - Access fields of [`union`]s
+///
+/// This seems rather arbitrary, why do I need a keyword for this ? The
+/// following section holds the answer.
```

Comment:
This is an odd thing to write. It seems to me this all would be easier if you *start* with "No matter what, Safe Rust can't cause Undefined Behavior". Then you can easily say "and thus, all operations that *could* cause UB must be banned from safe Rust. they are only available in Unsafe Rust. concretely, this affects the following operations:"

Reply:
Much better than what I had, thank you !